audisp-tacplus.c

```diff
@@ -1,652 +1,582 @@
-     * We use physical record number because we may search around and
-     * move the cursor accidentally skipping a record.
-     */
-    while(auparse_goto_record_num(au, num) > 0) {
-	type = auparse_get_type(au);
-	/*
-	 * we are only writing TACACS account records for syslog exec
-	 * records.  login, etc. are handled through pam_tacplus
-	 */
-	switch(type) {
-	    case AUDIT_SYSCALL:
-	    case AUDIT_ANOM_ABEND:
-		get_acct_record(au, type);
-		break;
-	    default:
-		// for doublechecking dump_whole_record(au);
-		break;
-	}
-	num++;
-    }
-}
+/*
+ * Copyright 2014, 2015, 2016, 2017 Cumulus Networks, Inc.  All rights reserved.
+ *   Based on audisp-example.c by Steve Grubb <sgrubb@redhat.com>
+ *     Copyright 2009 Red Hat Inc., Durham, North Carolina.
+ *     All Rights Reserved.
+ *
+ *   TACACS+ work based on pam_tacplus.c
+ *     Copyright (C) 2010, Pawel Krawczyk <pawel.krawczyk@hush.com> and
+ *     Jeroen Nijhof <jeroen@jeroennijhof.nl>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ *
+ * Author: olson@cumulusnetworks.com>
+ *
+ * This audisp plugin is used for TACACS+ accounting of commands
+ * being run by users known to the TACACS+ servers
+ * It uses libsimple_tacacct to communicate with the TACACS+ servers
+ * It uses the same configuration file format as the libnss_tacplus
+ * plugin (but uses the file /etc/audisp/audisp-tacplus.conf to
+ * follow the audisp conventions).
+ *
+ * You can test it by running commands similar to:
+ *   ausearch --start today --raw > test.log
+ *   ./audisp-tacplus < test.log
+ *
+ * Excluding some init/destroy items you might need to add to main, the
+ * event_handler function is the main place that you would modify to do
+ * things specific to your plugin.
+ *
+ */
+
+#define _GNU_SOURCE
+
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <pwd.h>
+#include <stdio.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <inttypes.h>
+#include <signal.h>
+#include <string.h>
+#include <ctype.h>
+#include <errno.h>
+#include <libaudit.h>
+#include <auparse.h>
+#include <unistd.h>
+#include <limits.h>
+
+#include <libtac/libtac.h>
+#include <trace.h>
+
+/* Tacacs+ support lib */
+#include <libtac/support.h>
+
+/* Remove password from user command*/

```

Comment:
>* [](http://example.com/codeflow?start=36&length=1)

Add one blank before `*`

Reply:
Fixed, also improve to handle escape char before split.

Reply:
The minor issue is still there.